packages/syft/src/syft/core/tensor/smpc/mpc_tensor.py

```diff
@@ -352,6 +353,78 @@ def __getattribute__(self, attr_name: str) -> Any:
 
         return object.__getattribute__(self, attr_name)
 
+    @staticmethod
+    def reshare(self: MPCTensor, parties: Iterable[Any]) -> MPCTensor:
+        """Reshare a given secret to a superset of parties.
+
+        Args:
+            parties(List[Any]): Input parites List.
+
+        Returns:
+            res_mpc(MPCTensor): Reshared MPCTensor.
+
+        Raises:
+            ValueError: If the input MPCTensor and input parties are same.
+        """
+        mpc_parties = set(self.parties)
+        parties = set(parties)
+        shape = self.shape
+        seed_shares = self.seed_shares
+        client_map = {share.client: share for share in self.child}
+        if mpc_parties == parties:
+            raise ValueError(
+                "Input parties for resharing are same as the input parties."
+            )
+
+        shares = [client_map.get(party) for party in parties]
+        for i, party in enumerate(parties):
```

Comment:
MAYBE:
```
res_shares = []
for share, party in zip(shares, parties):
    res_shares.append(generate_przs)
```

Reply:
Since it is dictionary, we automatically set `None` george if key not available, which automatically fits for the PRZS check in `ShareTensor`,I left it like that.

Reply:
It autmatically choose a Zero Tensor in ShareTensor code.

Reply:
It is actually , ``shares[i]`` george, if that part confuses, black formatted it that way :sweat: